### PR TITLE
Pinning attempt third

### PR DESCRIPTION
**Pull request by tunnell**
Don't merge till I say. Using TravisCI.

This is trying to pin the versions of dependencies to make for more stable testing and reproducibility.

### CODE REVIEW

**Comment by feigaodm** [comment]
well, first you have to make the test pass :)

**Comment by tunnell** [comment]
The following were autogenerated by pyup and mea culpa:

#687 #688 #689 #690 #691 #692 #693 #694 #695 #696 #697 #698 #699 #700

I closed and deleted branches

File: setup.py [hunk 1]
```diff
@@ -19,12 +19,12 @@
...
 # so check for h5py presence manually, and remove it from requirements if already found.
-try:
-    import h5py
-except ImportError:
-    pass
-else:
-    del requirements[requirements.index('h5py')]
+#try:
+#    import h5py
+#except ImportError:
```
Comment (tunnell) [inline setup.py]: __[Flake8]__

[E265] block comment should start with '# '

File: setup.py [hunk 2]
```diff
@@ -19,12 +19,12 @@
 
 # For some reason h5py is often not seen by pip if it was installed by conda...
 # so check for h5py presence manually, and remove it from requirements if already found.
-try:
-    import h5py
-except ImportError:
-    pass
-else:
-    del requirements[requirements.index('h5py')]
+#try:
```
Comment (tunnell) [inline setup.py]: __[Flake8]__

[E265] block comment should start with '# '

File: setup.py [hunk 3]
```diff
@@ -19,12 +19,12 @@
...
+#    import h5py
+#except ImportError:
+#    pass
+#else:
```
Comment (tunnell) [inline setup.py]: __[Flake8]__

[E265] block comment should start with '# '

**Comment by tunnell** [comment]
See #701

**Pull request closed without merging**